packages/trader/src/AppV2/Components/TradeParameters/Stake/stake.tsx

```diff
@@ -53,6 +54,19 @@ const Stake = observer(({ is_minimized }: TTradeParametersProps) => {
         ? Number(v2_params_initial_values.stake)
         : available_contract_types?.[contract_type]?.config?.default_stake;
 
+    useEffect(() => {
+        if (client_currency !== currency) {
+            onChange({ target: { name: 'currency', value: client_currency } });
+        }
+    }, [client_currency]);
+
+    useEffect(() => {
+        if (isLoggedOutV2) {
+            onChange({ target: { name: 'amount', value: default_stake_value } });
+            resetIsLoggedOutV2();
+        }
+    }, [isLoggedOutV2]);
```

Comment:
Can't we reuse existing `is_logged_in`?

Reply:
Also, not sure if it was a part of your card, but for log out users currency is not USD
<img width="449" alt="Screenshot 2024-11-06 at 1 46 12 PM" src="https://github.com/user-attachments/assets/fde4c314-2750-47e9-b9b9-6c23e2b49bff">

Reply:
the condition if we use `is_logged_in` . For logged_out user it will reset the stake on each page refresh . we don't have anything to know the exact event when user has clicked logout . that;s why I had to create it .

Reply:
https://github.com/user-attachments/assets/a6c483a1-b91c-4598-8aff-d5b42d1c1806

Reply:
Does it has to be USD , for staging and production it is AUD .

Reply:
What about case in the video? Selected BTC account -> log out -> note stake behaviour